packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/timing/TimingModuleTest.kt

```diff
@@ -0,0 +1,265 @@
+    mTimingModule.createTimer(41.0, 1.0, 0.0, true)
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(mJSTimersMock)
+    mTimingModule.onHeadlessJsTaskFinish(42)
+    stepChoreographerFrame()
+    verifyNoMoreInteractions(mJSTimersMock)
+  }
+
+  @Test
+  fun testHeadlessJsTaskInForeground() {
+    mTimingModule.onHostResume()
+    mTimingModule.onHeadlessJsTaskStart(42)
+    mTimingModule.createTimer(41.0, 1.0, 0.0, true)
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(mJSTimersMock)
+    mTimingModule.onHeadlessJsTaskFinish(42)
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(mJSTimersMock)
+    mTimingModule.onHostPause()
+    verifyNoMoreInteractions(mJSTimersMock)
+  }
+
+  @Test
+  fun testHeadlessJsTaskIntertwine() {
+    mTimingModule.onHostResume()
+    mTimingModule.onHeadlessJsTaskStart(42)
+    mTimingModule.createTimer(41.0, 1.0, 0.0, true)
+    mTimingModule.onHostPause()
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(mJSTimersMock)
+    mTimingModule.onHostResume()
+    mTimingModule.onHeadlessJsTaskFinish(42)
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(mJSTimersMock)
+    mTimingModule.onHostPause()
+    stepChoreographerFrame()
+    verifyNoMoreInteractions(mJSTimersMock)
+  }
+
+  @Test
+  fun testSetTimeoutZero() {
+    mTimingModule.createTimer(100.0, 0.0, 0.0, false)
+    verify(mJSTimersMock).callTimers(JavaOnlyArray.of(100.0))
+  }
+
+  @Test
+  fun testActiveTimersInRange() {
+    mTimingModule.onHostResume()
+    assertThat(mTimingModule.hasActiveTimersInRange(100)).isFalse
+    mTimingModule.createTimer(41.0, 1.0, 0.0, true)
+    assertThat(mTimingModule.hasActiveTimersInRange(100)).isFalse // Repeating
+    mTimingModule.createTimer(42.0, 150.0, 0.0, false)
+    assertThat(mTimingModule.hasActiveTimersInRange(100)).isFalse // Out of range
+    assertThat(mTimingModule.hasActiveTimersInRange(200)).isTrue // In range
+  }
+
+  @Test
+  @Ignore("Failing due to failed invocation verification") // TODO T13905097
+  fun testIdleCallback() {
+    mTimingModule.onHostResume()
+    mTimingModule.setSendIdleEvents(true)
+    stepChoreographerFrame()
+    verify(mJSTimersMock).callIdleCallbacks(SystemClock.currentTimeMillis().toDouble())
+  }
+
+  private class PostFrameCallbackHandler : Answer<Unit> {
+
+    private var mFrameCallback: FrameCallback? = null
+
+    override fun answer(invocation: InvocationOnMock) {
+      invocation.arguments[1]?.let { mFrameCallback = it as FrameCallback }
+    }
+
+    fun getAndResetFrameCallback(): FrameCallback? {
+      val callback = mFrameCallback
+      mFrameCallback = null
+      return callback
+    }
+  }
+
+  private class PostFrameIdleCallbackHandler : Answer<Unit> {
```

Comment:
Those two classes are the same. Can we delete one of the two and make sure we instantiate two of them instead